build.gradle

```diff
@@ -138,6 +141,15 @@ subprojects {
       precommit.dependsOn 'spotlessJavaCheck'
     }
   }
+  //copy the rest api spec to a global location and add to the test classpath
+  project.tasks.withType(RestIntegTestTask) {
```

Comment:
The `RestTestPlugin` creates a `integTest` task of type `RestIntegTestTask` task. But so does the `PluginBuildPlugin` (esplugin) (not sure why it explicitly creates it instead of applying the `RestTestPlugin`).

Additionally, many of the projects create custom tasks of type `RestIntegTestTask`. The requirement for the `RestTestRunnerTask` which is created by the `RestIntegTestTask`.  

Actually, I think this should actually be `withType(RestTestRunnerTask)` since a couple project create that directly.

Reply:
We aren't configuring the task, we are adding a `testRuntime` dependency to the _project_. _All_ test tasks inherit that classpath so there is no need to couple this to the existence of a `RestIntegTestTask`.

Reply:
Based on your comment below

> We should put the reliance on rest specs in an explicit plugin so that if you are using YAML tests and need those specs, you must apply the plugin.

I can move this to the `RestTestPlugin`